File: fyp-backend/OnboardingWFMSApi/OnboardingWFMSApi.DataModels/DTOs/WorkflowInstanceDTO.cs
using OnboardingWFMSApi.DataModels.Payloads;
using OnboardingWFMSApi.DataModels.Tables.Workflows;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace OnboardingWFMSApi.DataModels.DTOs
{
    public class WorkflowInstanceDTO : WorkflowInstanceTable
    {
        [JsonPropertyName("workflowTemplate")]
        public WorkflowTemplateDTO WorkflowTemplate { get; set; }
        [JsonPropertyName("completedTasks")]
        public int CompletedTasks { get; set; }
        /// <summary>
        /// Either PREFLOW, MAINFLOW or COMPLETED 
        /// </summary>
        [JsonPropertyName("status")]
        public string Status { get; set; }
        [JsonPropertyName("onboardingEmployeeDetails")]
        public OnboardingEmployeeDetailsDTO OnboardingEmployeeDetails { get; set; }
    }
}

File: fyp-frontend/src/models/DTOs/WorkflowInstanceDTO.ts
import OnboardingEmployeeDetailsDTO from "./OnboardingEmployeeDetailsDTO";
import WorkflowTemplateDTO from "./WorkflowTemplateDTO";

export default interface WorkflowInstanceDTO {
  workflowTemplate: WorkflowTemplateDTO;
  completedTasks: number;
  status: string;
  id: string;
  workflowTemplateId: string;
  supervisorAccountId: string;
  creationTimestamp: Date;
  onboardingEmployeeDetails: OnboardingEmployeeDetailsDTO | null;
}